Library/result.h

```diff
@@ -0,0 +1,19 @@
+/**
+ * @file
+ * @brief Result enum の定義
+ */
+#ifndef RESULT_H_
+#define RESULT_H_
+
+/**
+ * @enum   RESULT
+ * @brief  Rust でいうところの Result<T, E> のような，汎用的な返り値用 enum
```

Comment:
@sksat 

https://github.com/ut-issl/c2a-core/pull/504#issuecomment-1451652130

この表記があってるかチェックほしい

Reply:
それはちょっと思って，気持ちは分かるけどどう逆立ちしても `Result<T, E>` のようではないんだよなとはなった．
書くとしたらこんなかんじかな

```suggestion
 * @brief  Rust でいうところの Result<(), ()> のような，汎用的な成功したかどうかだけの返り値用 enum
```

Reply:
なおした